crates/symbolicator/src/services/symbolication/module_lookup.rs

```diff
@@ -82,17 +82,22 @@ impl ModuleLookup {
 
         modules.sort_by_key(|entry| entry.object_info.raw.image_addr.0);
 
-        // Ignore the name `dedup_by`, I just want to iterate over consecutive items and update
-        // some.
-        modules.dedup_by(|entry2, entry1| {
-            // If this underflows we didn't sort properly.
-            let size = entry2.object_info.raw.image_addr.0 - entry1.object_info.raw.image_addr.0;
-            if entry1.object_info.raw.image_size.unwrap_or(0) == 0 {
-                entry1.object_info.raw.image_size = Some(size);
+        // back-fill the `image_size` in case it is missing (or 0), so that it spans up to the
+        // next image
```

Comment:
If all of the images are already sorted by their `image_addr`, wouldn't there just be a small clump of zeroed images at the beginning of the list? If there are several zero-sized images in that clump, are they deterministically sorted? What are some of the possible side-effects of extending the sizes of images to fill in space they didn't claim to occupy to begin with?

Reply:
> If all of the images are already sorted by their image_addr, wouldn't there just be a small clump of zero-sized images at the beginning of the list?

I don't understand this question. How does a size of 0 relate to sorting by image_addr?

Reply:
I think "zero-sized" here refers to the image_size field containing zero due to us not knowing how big the image is, not it *actually* being size zero. But I may be misunderstanding this..